free5gc-operator/controllers/smfdeployment_controller.go

```diff
@@ -0,0 +1,648 @@
+		condition.Status = metav1.ConditionFalse
+		condition.Reason = "MinimumReplicasNotAvailable"
+		condition.Message = "SMFDeployment pod(s) is(are) starting."
+		condition.LastTransitionTime = metav1.Now()
+	} else if lastDeploymentStatus.Type == appsv1.DeploymentReplicaFailure {
+		condition.Type = string(workloadv1alpha1.Stalled)
+		condition.Status = metav1.ConditionFalse
+		condition.Reason = "MinimumReplicasNotAvailable"
+		condition.Message = "SMFDeployment pod(s) is(are) failing."
+		condition.LastTransitionTime = metav1.Now()
+	}
+
+	smfStatus.Conditions = append(smfStatus.Conditions, condition)
+
+	return smfStatus, true
+}
+
+//+kubebuilder:rbac:groups=workload.nephio.org,resources=smfdeployments,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=workload.nephio.org,resources=smfdeployments/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=workload.nephio.org,resources=smfdeployments/finalizers,verbs=update
+//+kubebuilder:rbac:groups=apps,resources=deployments,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=apps,resources=deployments/status,verbs=get
+//+kubebuilder:rbac:groups="",resources=pods,verbs=get;list;watch
+//+kubebuilder:rbac:groups="",resources=configmaps,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups="k8s.cni.cncf.io",resources=network-attachment-definitions,verbs=get;list;watch
+//+kubebuilder:rbac:groups=core,resources=events,verbs=create;patch
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the SMFDeployment object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.14.1/pkg/reconcile
+func (r *SMFDeploymentReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := log.FromContext(ctx).WithValues("SMFDeployment", req.NamespacedName)
+
+	smfDeploy := &workloadv1alpha1.SMFDeployment{}
+	err := r.Client.Get(ctx, req.NamespacedName, smfDeploy)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			log.Info("SMFDeployment resource not found. Ignoring since object must be deleted")
+			return reconcile.Result{}, nil
+		}
+		log.Error(err, "Error: failed to get SMFDeployment")
+		return reconcile.Result{}, err
+	}
+
+	namespace := smfDeploy.ObjectMeta.Namespace
+	// see if we are dealing with create or update
+	cmFound := false
+	configmapName := smfDeploy.ObjectMeta.Name + "-smf-configmap"
+	currConfigmap := &apiv1.ConfigMap{}
+	if err := r.Client.Get(ctx, types.NamespacedName{Name: configmapName, Namespace: namespace}, currConfigmap); err == nil {
+		cmFound = true
+	}
+
```

Comment:
are we swallowing errors? Or are they not errors but just means we have to create not update?